app/code/Magento/MediaGallerySynchronization/Model/Filesystem/FileInfo.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGallerySynchronization\Model\Filesystem;
+
+/**
+ * Internal class wrapping \SplFileInfo
+ *
+ * @SuppressWarnings(PHPMD.TooManyFields)
+ */
+class FileInfo extends \SplFileInfo
+{
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string
+     */
+    private $extension;
+
+    /**
+     * @var $basename
+     */
+    private $basename;
+
+    /**
+     * @var string
+     */
+    private $pathname;
+
+    /**
+     * @var int
+     */
+    private $inode;
+
+    /**
+     * @var int
+     */
+    private $size;
+
+    /**
+     * @var int
+     */
+    private $owner;
+
+    /**
+     * @var int
+     */
+    private $group;
+
+    /**
+     * @var int
+     */
+    private $aTime;
+
+    /**
+     * @var int
+     */
+    private $mTime;
+
+    /**
+     * @var int
+     */
+    private $cTime;
+
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var false|string
+     */
+    private $realPath;
+
+    /**
+     * FileInfo constructor.
+     * @param string $file_name
+     * @param string $path
+     * @param string $filename
+     * @param string $extension
+     * @param string $basename
+     * @param string $pathname
+     * @param int $inode
+     * @param int $size
+     * @param int $owner
+     * @param int $group
+     * @param int $aTime
+     * @param int $mTime
+     * @param int $cTime
+     * @param string $type
+     * @param false|string $realPath
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     */
+    public function __construct(
```

Comment:
In case of composition you will need only 1 argument - path. Then you will create an SplFileInfo object an proxy all required methods